sdks/python/apache_beam/transforms/util.py

```diff
@@ -573,3 +575,29 @@ def expand(self, pcoll):
             | 'AddRandomKeys' >> Map(lambda t: (random.getrandbits(32), t))
             | ReshufflePerKey()
             | 'RemoveRandomKeys' >> Map(lambda t: t[1]))
+
+
+@typehints.with_input_types(T)
+@typehints.with_output_types(T)
+class MustFollow(PTransform):
+  """Pass-through input, but enforces deferred processing.
+
+  Example usage:
+    # Ensure that output dir is created before attempting to write output files.
+    output_dir_ready = pipeline | beam.Create([output_path]) | CreateOutputDir()
+    output_pcoll | MustFollow(output_dir_ready) | WriteOutput()
+  """
+
+  def __init__(self, to_wait_for):
+    super(MustFollow, self).__init__()
+    self._to_wait_for = to_wait_for
+
+  def map_to_empty(self, x):
+    return []
+
+  def wait(self, x, unused_empty_dummy_side_input):
+    return x
+
+  def expand(self, pcoll):
+    empty_dummy = self._to_wait_for | 'MapToEmpty' >> FlatMap(self.map_to_empty)
```

Comment:
Having an unused side input will only work in runners that process a side input completely before processing its consumer. This doesn't hold for unbounded pipelines for all known Apache Beam runners.

Reply:
I'm not sure what semantics makes the most sense for an unbounded pipeline, but certainly open to suggestions for an alternative approach.

Reply:
I'm not sure this holds for bounded pipelines for all runners. I think you'll want to represent "must follows" as a new relationship between PTransforms and propose it on the dev@beam.apache.org mailing list. This way runners who can't honor it or if it doesn't make sense for the pipeline can reject it.

Here is an example of a similar proposal in this space of runner supported features: https://docs.google.com/document/d/117yRKbbcEdm3eIKB_26BHOJGmHSZl1YNoF0RqWGtqAM/edit#heading=h.xgjl2srtytjt

Reply:
I have added you to dev@beam.apache.org mailing list to figure this out.